utils/rpms/daos.spec

```diff
@@ -592,6 +592,11 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a shim package
 
 %changelog
+* Tue Sep 03 2024 Tomasz Gromadzki <tomasz.gromadzki@intel.com> 2.7.100-6
```

Comment:
why is this needed?  I don't see any new libs or anything

Reply:
While I generally agree with what @jolivier23 is asking here (when there are no other `daos.spec` file changes, the `Version:`/`Release:`/`%changelog` should not need updating and similarly for `debian/changelog` if no other files in `debian/` are updated), the added comment makes me wonder if we have a breaking change in PMDK with enabling NDCTL.  I don't think we should, otherwise the downstream testing for the PMDK update that enables NDCTL should have caught it if we did.

But what are the daos changes here doing exactly?  Are they purely an optional/optimization for a PMDK with NDCTL enabled?  Most importantly will they work with both a PMDK with NDCTL enabled and one without it enabled?  Does DAOS previous to these changes (i.e. current `master`) also work with both PMDKs?

Reply:
> why is this needed? I don't see any new libs or anything

It is the first version of the DAOS that will be able to work with PMDK with NDCTL enabled (after a very long break !!!).
It must be done BEFORE PMDK package will be equipped with NDCTL (see PR documentation: https://github.com/daos-stack/daos/pull/14371#issue-2296987020)

>  if we have a breaking change in PMDK with enabling NDCTL. I don't think we should, otherwise the downstream testing for the PMDK update that enables NDCTL should have caught it if we did

Turning on NDCTL in the PMDK package is crashing the system with the existing (master, 2.6) implementation. It has been detected by our CI. To solve the problem both the PMDK and the DAOS must be fixed.

> But what are the daos changes here doing exactly? Are they purely an optional/optimization for a PMDK with NDCTL enabled? 

The majority of fixes have been implemented inside the PMDK but the DAOS implementation (configuration) MUST also be adjusted to work with NDCTL.

> Most importantly will they work with both a PMDK with NDCTL enabled and one without it enabled? 

Yes, DAOS will work with and without NDCTL when these changes are applied.

> Does DAOS previous to these changes (i.e. current master) also work with both PMDKs?

No, the engine will crash if PMDK with NDCTL is used.

Reply:
yes

Reply:
I still don't quite understand why you need the spec change before that.   Why not defer the spec change until updating PMDK?

Reply:
First, we have to release DOAS that can use the new PMDK w/ NDCTL. Next, we release a new PMDK package and do not need the next DAOS release to use it. 
Eventually, we need to update DAOS itself (`build.config`) to use the new PMDK version for a local DAOS build, but this is a separate step.